docs/pages/application-access/guides/tcp.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Use Teleport for accessing plain TCP apps
+description: How to configure Teleport for accessing plain TCP apps.
+---
+
+# TCP Apps Access (Preview)
+
+Teleport can provide access to any TCP-based application. This allows users to
+connect to applications which Teleport doesn't natively support such as SMTP
+servers or databases not yet natively supported in Database Access.
+
+## Prerequisites
+
+You will need a running Teleport cluster, either self-hosted or in Teleport Cloud.
+We'll assume that you followed the [Getting Started](../getting-started.mdx)
+guide or the general [App Access Usage](./connecting-apps.mdx) guide to install
+Teleport and setup an Application Service.
+
+## Step 1/3. Configure TCP app
+
+Include your TCP application in the `teleport.yaml` configuration of Teleport
+Application Service:
+
+```yaml
+app_service:
+  enabled: "yes"
+  apps:
+  - name: "tcp-app"
+    uri: tcp://localhost:9876
+    labels:
+      app: tcp-app
+```
+
+Note that the URI scheme must be `tcp://` in order for Teleport to recognize
+this as a TCP application.
+
+You can also register the application [dynamically](./dynamic-registration.mdx)
+using `tctl create`.
+
+## Step 2/3. Start app proxy
+
+Log into your Teleport cluster and view available applications:
+
+```code
+$ tsh login --proxy=teleport.example.com
+$ tsh app ls
+Application Description   Type Public Address                   Labels
+----------- ------------- ---- -------------------------------- -----------
+tcp-app                   TCP  tcp-app.root.gravitational.io    app=tcp-app
+```
+
+Your TCP application should show up and be denoted with a `TCP` type.
+
+Now log into the application:
+
+```code
+$ tsh app login tcp-app
+Logged into TCP app tcp-app. Start the local TCP proxy for it:
+
+  tsh proxy app tcp-app
+
+Then connect to the application through this proxy.
+```
+
+Before you can connect to the application, you need to start a local proxy for
```

Comment:
If this step comes before the previous command, we should put it first. Otherwise, a user running each command in sequence will hit an error at `tsh app login`.